CodeU_Assignement_5/src/codeu_assignement_5/kemUnknownLanguage.java

```diff
@@ -0,0 +1,177 @@
+        {
+            if (!dict.contains(w.charAt(i)))
+            {
+                return i;
+            }
+        }
+        
+        return w.length();
+    }
+    
+    /*
+     * Input: the original set of words, current word we are working on, its prefix size
+     * Output: the set of the words with this prefix
+    */
+    private String[] getWordsWithSamePrefix(String[] alienWords, int curr, int prefixSize)
+    {
+        ArrayList<String> tempDict = new ArrayList();
+        tempDict.add(alienWords[curr]);
+        
+        String prefix = alienWords[curr].substring(0, prefixSize);
+        
+        /* Forward search */
+        for (int i=curr+1; i<alienWords.length; i++)
+        {
+            if (!alienWords[i].startsWith(prefix))
+            {
+                break;
+            }
+            
+            tempDict.add(alienWords[i]);
+        }
+        
+        /* Backward search */
+        for (int i=curr-1; i >= 0; i--)
+        {
+            if (!alienWords[i].startsWith(prefix))
+            {
+                break;
+            }
+            
+            tempDict.add(0,alienWords[i]); 
+        }
+        
+        String[] ret = new String[tempDict.size()];
+        for (int i=0; i < tempDict.size(); i++)
+        {
+            ret[i] = tempDict.get(i);
+        }
+        
+        return ret;
+    }
+    
+    /*
+     * Get two dictionary with common latters and merge them into the first dict,
+     * if none, cannot know, add all these cases later on
+    */
+    private void mergeDict(List<Character> dict1, List<Character> dict2)
+    {
+        int from2 = 0;  
+        for (int i=0; i < dict2.size(); i++)
+        {
+            if (dict1.contains(dict2.get(i)))
+            {
+                /* Find common point, merge here! */
+                dict1.addAll(dict1.indexOf(dict2.get(i)), dict2.subList(from2, i));
+                from2 = i+1;
+            }        
+        }
+    }
+    
+    private boolean addToDictByPrefix(String[] alienWords, List<Character> dict)
+    {
+        boolean hasReachMaxPrefix = true;
+        for (int i=0; i < alienWords.length; i++)
+        {
+            int prefixSize = getSizeOfPrefix(alienWords[i], dict);
+            if (prefixSize != alienWords[i].length())
+            {
+                String[] sublistWords = getWordsWithSamePrefix(alienWords, i, prefixSize);
+                if (sublistWords.length > 1)
+                {
+                    List<Character> tempDict = getBasicAlphaBeit(sublistWords, prefixSize);
+                    if (!tempDict.isEmpty())
+                    {
+                        int dictSize = dict.size();
+                        mergeDict(dict,tempDict);
+                        if (dictSize != dict.size())
+                        {
+                            hasReachMaxPrefix = false;
+                        }
+                    }
+                }
+            }
+        }
+        

```

Comment:
No sure we shall continue for the third iteration... 
Also what happen in this case:
["ART", "RAT", "CTTA", "CTTT", "CTA", "CRA"]
I think the "legal input" was a bit of an open question here...